api/src/Page/Sample.php

```diff
@@ -1928,8 +1925,11 @@ function _update_sample()
             $dcps = $this->db->pq("SELECT dataCollectionPlanId FROM BLSample_has_DataCollectionPlan
                                 WHERE blSampleId = :1", array($this->arg('sid')));
 
-            if (sizeof($dcps)) {
-                foreach ($dcps as $dcp) {
+            if(sizeof($dcps)) {
+                $maxLocation = $this->_get_current_max_dcp_plan_order($this->args['CONTAINERID']);
+                $maxLocation = array_key_exists('CONTAINERID', $this->args) ? $this->_get_current_max_dcp_plan_order($this->args['CONTAINERID']) : array(0);
+                
```

Comment:
```suggestion
                $maxLocation = sizeof($maxLocation) ? $maxLocation : -1;
```
Don't we need this line?